cookiecutter/environment.py

```diff
@@ -7,6 +7,31 @@
 from .exceptions import UnknownExtension
 
 
+class CustomEnvironmentArgsMixin(object):
```

Comment:
While this is convenient, it seems like the cookiecutter prefers users using jinja2 extensions to extend functionality.

You can write a simple extension like this for your use case

```
class CustomDelimitersExtension(Extension):
    def __init__(self, environment):

        super(CustomDelimitersExtension, self).__init__(environment)
        environment.variable_start_string = "[["
        environment.variable_end_string = "]]"
```

Reply:
I would probably remove the addition of the mixin from this PR to make it easier to accept.
I think the key point you covered, is passing the env variable to `find_template`. Extensions are loaded when env is instantiated, so your variable delimiters would be taken into account.